common/djangoapps/util/password_policy_validators.py

```diff
@@ -1,322 +1,467 @@
+    for validator in password_validators:
+        text = validator.get_instruction_text()
+        if text:
+            if isinstance(validator, MinimumLengthValidator):
+                length_instruction = text
+            else:
+                complexity_instructions.append(text)
+    if complexity_instructions:
+        return _('Your password must contain {length_instruction}, including {complexity_instructions}.').format(
+            length_instruction=length_instruction,
+            complexity_instructions=' & '.join(complexity_instructions)
+        )
+    else:
+        return _('Your password must contain {length_instruction}.'.format(length_instruction=length_instruction))
 
 
-def password_min_length():
+def password_validators_restrictions(password_validators=None):
     """
-    Returns minimum required length of a password.
-    Can be overridden by site configuration of PASSWORD_MIN_LENGTH.
+    Return a dictionary of complexity restrictions to be used by mobile users on
+    the registration form
     """
-    min_length = getattr(settings, 'PASSWORD_MIN_LENGTH', None)
-    if min_length is None:
-        return 2  # Note: This default is simply historical
-    return min_length
+    complexity_restrictions = {}
+    if password_validators is None:
+        password_validators = get_default_password_validators()
+    for validator in password_validators:
+        key, value = validator.get_restriction()
+        if key:
+            complexity_restrictions[key] = value
+    return complexity_restrictions
 
 
-def password_max_length():
+def unicode_check(password):
     """
-    Returns maximum allowed length of a password. If zero, no maximum.
-    Can be overridden by site configuration of PASSWORD_MAX_LENGTH.
+    Encodes the password in utf8 if it is not already.
+
+    Returns the password.
     """
-    # Note: The default value here is simply historical
-    max_length = getattr(settings, 'PASSWORD_MAX_LENGTH', None)
-    if max_length is None:
-        return 75  # Note: This default is simply historical
-    return max_length
+    if not isinstance(password, text_type):
+        try:
+            # some checks rely on unicode semantics (e.g. length)
+            password = text_type(password, encoding='utf8')
+        except UnicodeDecodeError:
+            # no reason to get into weeds
+            raise ValidationError([_('Invalid password.')])
+    return password
 
 
-def password_complexity():
+def _validate_condition(password, fn, min_count):
     """
-    :return: A dict of complexity requirements from settings
+    Validates the password using the given function. This is performed by
+    iterating through each character in the password and counting up the number
+    of characters that satisfy the function.
+
+    Parameters:
+        password (str): the password
+        fn (str): the function to be tested against the string.
+            Examples are: 'isalpha', 'isnumeric', etc.
```

Comment:
Another way to go would be allow an actual function input, which is what I would normally assume a `fn` parameter was.

It requires a little more code when calling for the common case:
`validate_condition(password, lambda c: c.isupper(), count)`

But also means that you aren't so limited in what you can check. So you could use it on the symbol and punctuation cases too.